common/types.ts

```diff
@@ -0,0 +1,107 @@
+export interface User {
+    email: string;
+    firstName: string;
+    lastName: string;
+    userId: number; 
+    gender: Gender;
+    school: string;
+    education: Education;
+    yearOfEducation: YearOfEducation;
+    ethnicities: Ethnicities;
+    shirtSize: ShirtSize;
+    applicationStatus: ApplicationStatus;
+    major: string;
+    minor: string;
+    resumeLink: string;
+    timeZone: string;
+    learningGoals: string;
+}
+
+export interface PortalState {
+    openDate: Date;
+    closeDate: Date;
+    maxAttendees: number;
+    numAttendees: number;
+    totalConfirmed: number;
+
+}
+
+enum Gender {
+    Nonbinary="Nonbinary", Female="Female", Male="Male", Unspecified="Unspecified"
+}
+
+enum Education {
+    HighSchool="HighSchool", Undergraduate="Undergraduate", Graduate="Graduate", 
+    Doctorate="Doctorate"
+}
+
+type YearOfEducation = 1 | 2 | 3 | 4 | 5 | "5+";
+
+enum Ethnicities {
+    "Indigenous/AlaskaNative"="Indigenous/AlaskaNative", Asian="Asian", "Black/AfricanAmerican"="Black/AfricanAmerican", 
+    "Hispanic/Latino"="Hispanic/Latino", "NativeHawaiian/PacificIslander" = "NativeHawaiian/PacificIslander", white = "white",
+    other = "other"
+}
+
+enum ShirtSize {
+    XSmall = "XSmall", Small = "Small", Medium = "Medium", Large = "Large", XLarge = "XLarge"
+}
+
+enum ApplicationStatus {
+    Unverified = "Unverified", 
+    "Incomplete/RegistrationOpen" = "Incomplete/RegistrationOpen",
+    "Incomplete/RegistrationClosed" = "Incomplete/RegistrationClosed",
+    "Submitted/RegistrationOpen" = "Submitted/RegistrationOpen",
+    "Submitted/RegistrationClosed" = "Submitted/RegistrationClosed",
+    "Admitted/Unconfirmed" = "Admitted/Unconfirmed", 
+    "Admitted/ConfirmedDeadlinePassed" = "Admitted/ConfirmedDeadlinePassed",
+    Waitlisted = "Waitlisted",
+    Confirmed = "Confirmed", 
+    Declined = "Declined"
+}
+
+enum QuestionType {
+    Checkboxes = "Checkboxes", 
+    ShortText = "ShortText", 
+    Dropdown = "Dropdown", 
+    LongText = "LongText"
+}
+
+type Question = Checkboxes | ShortText | Dropdown | LongText;
+
+interface IQuestion {
+    content: string;
+    id: number;
+    required: boolean;
+}
+
+interface Checkboxes extends IQuestion {
+    type : QuestionType.Checkboxes;
+    options : Array<{name: string}>;
+    maxNumber : number;
+    minNumber : number;
+}
+
+interface ShortText extends IQuestion {
+    type : QuestionType.ShortText;
+    maxLength : number;
+    minLength : number;
+}
+
+interface Dropdown extends IQuestion {
+    type : QuestionType.Dropdown;
+    options : Array<{name: string}>;
+
+}
+
+interface LongText extends IQuestion {
+    type : QuestionType.LongText;
+    maxLength : number;
+    minLength : number;
+}
+
+interface QuestionResponse {
+    userID: number;
+    question: IQuestion;
+    response : Array<string>; // <- can change
```

Comment:
Wouldn't a response just be one string, not a list of strings?

Reply:
yeah we're still working on this – planning on changing it to a union type (eg for dropdown, a single string, but for checkboxes, a list of strings, etc)

Reply:
Gotcha, forgot that multi-select answers exist...I think if Dropdowns are just lists with one item that's fine too (obviously union would be more "correct")

Reply:
Oh I see, just to make sure so we would write a type Response = string | Array<string> and then in questionResponse object line 106 would now be response: Response

Reply:
Would we need to use Response elsewhere? If not, imo it's more readable to just define the union in QuestionResponse. Also I think it would still be Array<String>

Reply:
I don't think we need Response anywhere else so we could leave 106 as it is?